### PR TITLE
Create install-collector.sh

### PR DESCRIPTION
**Pull request by shyamvalsan**
This script does a 1 line installation of community maintained collectors from netdata/community github repo

Test results:

```bash
cm@shyam-vm1:/etc/netdata$ sudo ./install-collector.sh python.d.plugin/clickhouse

You are about to install a collector that is NOT developed or maintained by Netdata. Would you like to proceed? [y/n]y

Installing collector from community repo...--2022-11-21 22:32:51--  https://raw.githubusercontent.com/netdata/community/main/collectors/python.d.plugin/clickhouse/clickhouse.chart.py
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.111.133, 185.199.110.133, 185.199.108.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.111.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 14588 (14K) [text/plain]
Saving to: ‘/usr/libexec/netdata/python.d/clickhouse.chart.py’

clickhouse.chart.py                                  100%[===================================================================================================================>]  14.25K  --.-KB/s    in 0.004s

2022-11-21 22:32:52 (3.56 MB/s) - ‘/usr/libexec/netdata/python.d/clickhouse.chart.py’ saved [14588/14588]

--2022-11-21 22:32:52--  https://raw.githubusercontent.com/netdata/community/main/collectors/python.d.plugin/clickhouse/clickhouse.conf
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.109.133, 185.199.108.133, 185.199.110.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.109.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1415 (1.4K) [text/plain]
Saving to: ‘/etc/netdata/python.d/clickhouse.conf’

clickhouse.conf                                      100%[===================================================================================================================>]   1.38K  --.-KB/s    in 0s

2022-11-21 22:32:52 (14.2 MB/s) - ‘/etc/netdata/python.d/clickhouse.conf’ saved [1415/1415]

"clickhouse: yes"

Successfully restarted Netdata agent

Installed clickhouse collector succesfully.

cm@shyam-vm1:/etc/netdata$ ls -lrt /usr/libexec/netdata/python.d/clickhouse.chart.py
-rw-r--r-- 1 root root 14588 Nov 21 22:32 /usr/libexec/netdata/python.d/clickhouse.chart.py
cm@shyam-vm1:/etc/netdata$ ls -lrt /etc/netdata/python.d/clickhouse.conf
-rw-r--r-- 1 root root 1415 Nov 21 22:32 /etc/netdata/python.d/clickhouse.conf
cm@shyam-vm1:/etc/netdata$ head /etc/netdata/python.d.conf
"clickhouse: yes"
cm@shyam-vm1:/etc/netdata$ date
Mon 21 Nov 2022 10:33:43 PM EET
```

### CODE REVIEW

**Comment by shyamvalsan** [comment]
@thiagoftsm @andrewm4894 @ilyam8 

A script that installs collectors from netdata/community repo.

File: utilities/install-collector.sh
Comment (thiagoftsm) [inline utilities/install-collector.sh:10]: Probably we need to create this script at `netdata/netdata`, because our `kickstart-startic` installs code at `/opt/`, so the default directory is `/opt/etc/netdata`, and as you can see in next code

https://github.com/netdata/netdata/blob/b8d50ecc953f7962587aaae1a793827faf4ce992/netdata-installer.sh#L479-L485

we add a prefix for path.

File: utilities/install-collector.sh
Comment (thiagoftsm) [inline utilities/install-collector.sh]: Considering this is an error, it is better to pass a number different of zero as argument to give information about error.

File: utilities/install-collector.sh
Comment (thiagoftsm) [inline utilities/install-collector.sh:53]: Like first comment, if we do not move this PR for `netdata/netdata`, we should either recognize where `netdata` is installed or allow users to give additional information about where `netdata` is installed.

File: utilities/install-collector.sh
Comment (thiagoftsm) [inline utilities/install-collector.sh]: Here  you are assuming that `wget` is always installed, unfortunately this is not true. Users can have only `curl` on their environment, please, take a look at next link

https://github.com/netdata/netdata/blob/b8d50ecc953f7962587aaae1a793827faf4ce992/packaging/installer/functions.sh#L109-L129

We should have something like this for this script.

File: utilities/install-collector.sh
Comment (thiagoftsm) [inline utilities/install-collector.sh:88]: There are distributions like :heart: Slackware Linux :heart: that do not have `systemd` installed, on these distributions we won't have neither `systemctl` nor  `systemd`, please, add commands to verify that commands are present before to run them.

File: utilities/install-collector.sh
Comment (thiagoftsm) [inline utilities/install-collector.sh:103]: @Ferroin considering that we are now having shell scripts on this repo, we need to add `shellcheck` to test PRs before to merge.

File: utilities/install-collector.sh
Comment (shyamvalsan) [reply utilities/install-collector.sh:10]: @thiagoftsm ideally, this script should be available on the agent node so users can just run it to install community collectors. Maybe once the review of this is done we can agree on where to move it in `netdata/netdata`?

File: utilities/install-collector.sh
Comment (shyamvalsan) [reply utilities/install-collector.sh:53]: Agree - I think we should move it to `netdata/netdata`

File: utilities/install-collector.sh
Comment (shyamvalsan) [reply utilities/install-collector.sh:88]: Updated. To keep it simple here's what I am doing. 

1. Check if systemctl is active
2. If systemctl is active then try restarting - If restart fails exit with error
3. If systemstl is not active then exit with error. 

I think this should be OK since after this exit all the user has to do is a manual restart, the rest is already done.

File: utilities/install-collector.sh
Comment (shyamvalsan) [reply utilities/install-collector.sh:103]: @thiagoftsm I have run shellcheck on this btw. Its all good.

File: utilities/install-collector.sh
Comment (thiagoftsm) [reply utilities/install-collector.sh:10]: All right, I suggest we use either [packaging](https://github.com/netdata/netdata/tree/master/packaging) or [collectors](https://github.com/netdata/netdata/tree/master/collectors). (ping @Ferroin )

File: utilities/install-collector.sh
Comment (thiagoftsm) [reply utilities/install-collector.sh:103]: Perfect, I will keep this unresolved waiting for @Ferroin answer.

File: utilities/install-collector.sh
Comment (shyamvalsan) [reply utilities/install-collector.sh]: Added

File: utilities/install-collector.sh
Comment (thiagoftsm) [reply utilities/install-collector.sh:88]: I think we are almost there, the unique issue remaining is the fact that `systemctl` is not installed in some systems:

```sh
bash-5.2# systemctl
bash: systemctl: command not found
bash-5.2# 
```

So we need to do here a test like you brought for `wget` and `curl`, after this the PR will be ready to be merged. :)

Comment (shyamvalsan) [reply utilities/install-collector.sh:88]: Done 😄

**Review by thiagoftsm — APPROVED** [review]
Thank you again @shyamvalsan 
Let us wait for @Ferroin answers and move this for `netdata/netdata`. :smile: .

**Comment by andrewm4894** [comment]
@shyamvalsan there should be a readme in utilities folder maybe that has some info on each script or at least a comment in the script showing example usage.

**Comment by andrewm4894** [comment]
fyi im going to try it like this: 

```bash
sudo wget -O /tmp/install-collector.sh https://raw.githubusercontent.com/netdata/community/main/utilities/install-collector.sh && sh /tmp/install-collector.sh charts.d.plugin/speedtest
```

**Comment by andrewm4894** [comment]
```
andrewmaguire@ml-demo-ml-enabled:~$ sudo wget -O /tmp/install-collector.sh https://raw.githubusercontent.com/netdata/community/main/utilities/install-collector.sh && sh /tmp/install-collector.sh charts.d.plugin/speedtest
--2022-11-24 12:06:02--  https://raw.githubusercontent.com/netdata/community/main/utilities/install-collector.sh
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.109.133, 185.199.108.133, 185.199.111.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.109.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 3865 (3.8K) [text/plain]
Saving to: ‘/tmp/install-collector.sh’

/tmp/install-collector.sh                         100%[=============================================================================================================>]   3.77K  --.-KB/s    in 0s      

2022-11-24 12:06:02 (30.2 MB/s) - ‘/tmp/install-collector.sh’ saved [3865/3865]

/tmp/install-collector.sh: 12: [: Illegal number: 
/tmp/install-collector.sh: 18: [: charts.d.plugin/speedtest: unexpected operator

You are about to install a collector that is NOT developed or maintained by Netdata. Would you like to proceed? [y/n]y

Installing collector from community repo.../tmp/install-collector.sh: 41: [: charts.d: unexpected operator
/tmp/install-collector.sh: 43: [: charts.d: unexpected operator
/tmp/install-collector.sh: 45: [: charts.d: unexpected operator
/tmp/install-collector.sh: 47: [: charts.d: unexpected operator
--2022-11-24 12:06:07--  https://raw.githubusercontent.com/netdata/community/main/collectors/charts.d.plugin/speedtest/speedtest.chart
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.109.133, 185.199.108.133, 185.199.111.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.109.133|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2022-11-24 12:06:07 ERROR 404: Not Found.

--2022-11-24 12:06:07--  https://raw.githubusercontent.com/netdata/community/main/collectors/charts.d.plugin/speedtest/speedtest.conf
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.109.133, 185.199.108.133, 185.199.111.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.109.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 470 [text/plain]
Saving to: ‘/etc/netdata/charts.d/speedtest.conf’

speedtest.conf                                    100%[=============================================================================================================>]     470  --.-KB/s    in 0s      

2022-11-24 12:06:07 (13.4 MB/s) - ‘/etc/netdata/charts.d/speedtest.conf’ saved [470/470]

tee: /etc/netdata/charts.d.conf: Permission denied
"speedtest: yes"

Unable to restart Netdata agent, please follow these instructions to restart manually: https://learn.netdata.cloud/docs/configure/start-stop-restart
/usr/bin/systemctl
```

**Comment by andrewm4894** [comment]
@shyamvalsan that does not look like it worked for me

**Comment by shyamvalsan** [comment]
@andrewm4894 you would need to run it as follows (bash instead of sh):

```bash
sudo wget -O /tmp/install-collector.sh https://raw.githubusercontent.com/netdata/community/main/utilities/install-collector.sh && sudo bash /tmp/install-collector.sh charts.d.plugin/speedtest
```

**Comment by andrewm4894** [comment]
got this now on same machine (maybe we could add option to clobber existing): 

```
andrewmaguire@ml-demo-ml-enabled:~$ sudo wget -O /tmp/install-collector.sh https://raw.githubusercontent.com/netdata/community/main/utilities/install-collector.sh && sudo bash /tmp/install-collector.sh charts.d.plugin/speedtest
--2022-11-24 12:46:29--  https://raw.githubusercontent.com/netdata/community/main/utilities/install-collector.sh
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.110.133, 185.199.108.133, 185.199.111.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.110.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 3865 (3.8K) [text/plain]
Saving to: ‘/tmp/install-collector.sh’

/tmp/install-collector.sh                         100%[=============================================================================================================>]   3.77K  --.-KB/s    in 0s      

2022-11-24 12:46:29 (26.7 MB/s) - ‘/tmp/install-collector.sh’ saved [3865/3865]


You are about to install a collector that is NOT developed or maintained by Netdata. Would you like to proceed? [y/n]y

Installing collector from community repo...

A Netdata collector by this name already exists on this system. Exiting...
```

running on another node fresh i get this: 

```
andrewmaguire@ml-demo-ml-disabled:~$ sudo wget -O /tmp/install-collector.sh https://raw.githubusercontent.com/netdata/community/main/utilities/install-collector.sh && sudo bash /tmp/install-collector.sh charts.d.plugin/speedtest
--2022-11-24 12:48:27--  https://raw.githubusercontent.com/netdata/community/main/utilities/install-collector.sh
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.109.133, 185.199.110.133, 185.199.111.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.109.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 3865 (3.8K) [text/plain]
Saving to: ‘/tmp/install-collector.sh’

/tmp/install-collector.sh                         100%[=============================================================================================================>]   3.77K  --.-KB/s    in 0s      

2022-11-24 12:48:27 (22.3 MB/s) - ‘/tmp/install-collector.sh’ saved [3865/3865]


You are about to install a collector that is NOT developed or maintained by Netdata. Would you like to proceed? [y/n]y

Installing collector from community repo...--2022-11-24 12:48:30--  https://raw.githubusercontent.com/netdata/community/main/collectors/charts.d.plugin/speedtest/speedtest.chart.sh
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.109.133, 185.199.110.133, 185.199.111.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.109.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 6498 (6.3K) [text/plain]
Saving to: ‘/usr/libexec/netdata/charts.d/speedtest.chart.sh’

speedtest.chart.sh                                100%[=============================================================================================================>]   6.35K  --.-KB/s    in 0s      

2022-11-24 12:48:30 (32.1 MB/s) - ‘/usr/libexec/netdata/charts.d/speedtest.chart.sh’ saved [6498/6498]

--2022-11-24 12:48:30--  https://raw.githubusercontent.com/netdata/community/main/collectors/charts.d.plugin/speedtest/speedtest.conf
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.109.133, 185.199.110.133, 185.199.111.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.109.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 470 [text/plain]
Saving to: ‘/etc/netdata/charts.d/speedtest.conf’

speedtest.conf                                    100%[=============================================================================================================>]     470  --.-KB/s    in 0s      

2022-11-24 12:48:31 (10.9 MB/s) - ‘/etc/netdata/charts.d/speedtest.conf’ saved [470/470]

"speedtest: yes"

Successfully restarted Netdata agent.

Installed speedtest collector succesfully.
```

**Comment by andrewm4894** [comment]
@shyamvalsan wonder what this is: 

```
andrewmaguire@ml-demo-ml-disabled:~$ grep 'speedtest' /var/log/netdata/error.log
/etc/netdata/charts.d.conf: line 1: speedtest: yes: command not found
/usr/libexec/netdata/plugins.d/../charts.d/speedtest.chart.sh: line 45: speedtest: command not found
/usr/libexec/netdata/plugins.d/../charts.d/speedtest.chart.sh: line 47: *8/1000: syntax error: operand expected (error token is "*8/1000")
```

**Comment by andrewm4894** [comment]
anyways - script seems to work - just used it for python clickhouse collector. This is cool, defo should add a note to docs somewhere about how to use it.

**Comment by shyamvalsan** [comment]
> @shyamvalsan wonder what this is:
> 
> ```
> andrewmaguire@ml-demo-ml-disabled:~$ grep 'speedtest' /var/log/netdata/error.log
> /etc/netdata/charts.d.conf: line 1: speedtest: yes: command not found
> /usr/libexec/netdata/plugins.d/../charts.d/speedtest.chart.sh: line 45: speedtest: command not found
> /usr/libexec/netdata/plugins.d/../charts.d/speedtest.chart.sh: line 47: *8/1000: syntax error: operand expected (error token is "*8/1000")
> ```

Hmm, I think I might know what that is, will fix it.

**Pull request merged**
